Userland/Libraries/LibGfx/Font/Emoji.cpp

```diff
@@ -58,18 +58,47 @@ Bitmap const* Emoji::emoji_for_code_point_iterator(Utf8CodePointIterator& it)
     struct EmojiAndCodePoints {
         Bitmap const* emoji;
         Span<u32> code_points;
+        u8 real_codepoint_length;
     };
     Vector<EmojiAndCodePoints, max_emoji_code_point_sequence_length> possible_emojis;
 
     // Determine all existing emojis for the longest possible ZWJ emoji sequence,
     // or until we run out of code points in the iterator.
-    for (size_t i = 0; i < max_emoji_code_point_sequence_length; ++i) {
+    bool last_codepoint_sequence_found = false;
+    for (u8 i = 0; i < max_emoji_code_point_sequence_length; ++i) {
         auto code_point = it.peek(i);
         if (!code_point.has_value())
             break;
-        code_points.append(*code_point);
-        if (auto const* emoji = emoji_for_code_points(code_points))
-            possible_emojis.empend(emoji, code_points);
+        // From https://unicode.org/reports/tr51/#Emoji_Implementation_Notes:
+        // > minimally-qualified or unqualified emoji zwj sequences may be handled
+        // > in the same way as their fully-qualified forms; the choice is up to
+        // > the implementation.
+        //
+        // Serenity chooses to treat minimally-qualified and unqualified
+        // emojis the same as fully-qualified emojis.
+        //
+        // So, if the emoji presentation specifier is found, just skip it in order
+        // to drop fully-qualified emojis down to their minimally-qualified or
+        // unqualified forms. This ensures that all forms are treated the same
```

Comment:
I get what you are doing but I think the comment can be a bit clearer:
1) This apply to emoji that is not ZWJ but has U+FE0F, correct?
2) Fully-qualified forms is needed for skin tones and similar, so we only treat minimally-qualified and unqualified emojis the same as fully-qualified emojis in regards to variation selector-16(U+FE0F), not other cases, correct?

Reply:
You're probably right, I'm likely being imprecise. I'll try to improve this after sleeping.

Note to self: this is the relevant Unicode page for the correct terminology: https://unicode.org/reports/tr51/#Definitions

Reply:
Updated, let me know what you think of it now.